tests/test_decoding.py

```diff
@@ -1242,7 +1242,7 @@ def test_decode_response(self) -> None:
                 coding_object=message,
                 param_dict={
                     "SID": sid,
-                    "matching_req_param": bytes([0xAB])
+                    "matching_req_param": 0xAB
```

Comment:
why?

Reply:
I thought that integers are more intuitive for matching request parameters, and they need to be decoded as integers anyway since they can be non-byte-aligned. maybe I should spin this change off...

Reply:
okay, I had a closer look: the code that is currently in `main` only supports byte-aligned matching request parameters. anyway, I reverted this change for now...